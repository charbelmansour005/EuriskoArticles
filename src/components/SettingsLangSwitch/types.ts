export type SettingsLangSwitchProps = {
  onToggleSwitch: () => void
  language: any
}

export type LangFeature = {
  englishDefinition: string
  frenchDefinition: string
}
