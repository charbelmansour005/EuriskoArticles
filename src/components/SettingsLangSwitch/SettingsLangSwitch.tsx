import {StyleSheet, Text, View} from 'react-native'
import {Switch} from 'react-native-paper'
import React, {Fragment} from 'react'
import {SettingsLangSwitchProps, LangFeature} from './types'
import {themeColors} from '../../helpers/themeColors'

const SettingsLangSwitch = ({
  ...props
}: SettingsLangSwitchProps): JSX.Element => {
  const LangFeature: LangFeature = {
    englishDefinition:
      'Changing the language will only affect that of the application and not the language which the articles are displayed in, the change will be system wide.',
    frenchDefinition:
      "Changer la langue n'affectera que celle de l'application et non la langue dans laquelle les articles sont affichés, le changement sera à l'échelle du système.",
  }

  return (
    <Fragment>
      <View style={styles.paperContainerFlex} testID="parent">
        <Text style={styles.BoldSmallNoMargin} testID="languageOptionFrench">
          {props?.language?.english ? `French` : `Francais`}
        </Text>
        <Switch
          color="silver"
          value={props?.language?.english}
          onValueChange={props?.onToggleSwitch}
          testID="switch"
        />
        <Text style={styles.BoldSmallNoMargin} testID="languageOptionEnglish">
          {' '}
          {props?.language?.english ? `English` : `Anglais`}
        </Text>
      </View>
      <View testID="languageExplanation">
        {props?.language?.english ? (
          <Text style={styles.languageExplanation}>
            {LangFeature.englishDefinition}
          </Text>
        ) : (
          <Text style={styles.languageExplanation}>
            {LangFeature.frenchDefinition}
          </Text>
        )}
      </View>
    </Fragment>
  )
}

export default SettingsLangSwitch

const styles = StyleSheet.create({
  BoldSmallNoMargin: {
    fontWeight: 'bold',
    fontSize: 13,
    color: themeColors.darkblue,
  },
  paperContainerFlex: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'space-around',
    alignContent: 'center',
    backgroundColor: 'white',
    margin: 10,
    padding: 10,
    borderRadius: 10,
    borderWidth: 0.5,
    borderColor: 'lightgray',
    flexDirection: 'row',
  },
  languageExplanation: {
    marginLeft: 20,
    fontSize: 13,
    marginBottom: 20,
    lineHeight: 20,
    color: 'black',
  },
})
