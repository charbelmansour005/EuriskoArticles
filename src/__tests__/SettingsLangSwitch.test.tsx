import React from 'react'
import renderer from 'react-test-renderer'
import SettingsLangSwitch from '../components/SettingsLangSwitch/SettingsLangSwitch'

test('renders correctly', async () => {
  const tree = renderer
    .create(
      <SettingsLangSwitch
        onToggleSwitch={function (): void {
          throw new Error('Function not implemented.')
        }}
        language={undefined}
      />,
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

test('parent view shows', async () => {
  const instance = renderer.create(
    <SettingsLangSwitch
      onToggleSwitch={function (): void {
        throw new Error('Function not implemented.')
      }}
      language={undefined}
    />,
  ).root
  expect(instance.findByProps({testID: 'parent'})).toBeTruthy()
})

test('french language options text shows', async () => {
  const instance = renderer.create(
    <SettingsLangSwitch
      onToggleSwitch={function (): void {
        throw new Error('Function not implemented.')
      }}
      language={undefined}
    />,
  ).root
  expect(instance.findByProps({testID: 'languageOptionFrench'})).toBeTruthy()
})

test('switch element shows', async () => {
  const instance = renderer.create(
    <SettingsLangSwitch
      onToggleSwitch={function (): void {
        throw new Error('Function not implemented.')
      }}
      language={undefined}
    />,
  ).root
  expect(instance.findByProps({testID: 'switch'})).toBeTruthy()
})

test('english language option text shows', async () => {
  const instance = renderer.create(
    <SettingsLangSwitch
      onToggleSwitch={function (): void {
        throw new Error('Function not implemented.')
      }}
      language={undefined}
    />,
  ).root
  expect(instance.findByProps({testID: 'languageOptionEnglish'})).toBeTruthy()
})

test('language explanation text shows', async () => {
  const instance = renderer.create(
    <SettingsLangSwitch
      onToggleSwitch={function (): void {
        throw new Error('Function not implemented.')
      }}
      language={undefined}
    />,
  ).root
  expect(instance.findByProps({testID: 'languageExplanation'})).toBeTruthy()
})
