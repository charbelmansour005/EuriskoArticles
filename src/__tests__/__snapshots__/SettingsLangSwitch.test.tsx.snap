// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsLangSwitch Component renders correctly 1`] = `
Array [
  <View
    style={
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "backgroundColor": "white",
        "borderColor": "lightgray",
        "borderRadius": 10,
        "borderWidth": 1,
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-around",
        "margin": 10,
        "padding": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 13,
          "fontWeight": "bold",
        }
      }
    >
      Francais
    </Text>
    <RCTSwitch
      accessibilityRole="switch"
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onTintColor="silver"
      style={
        Object {
          "height": 31,
          "width": 51,
        }
      }
      tintColor="silver"
      value={false}
    />
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 13,
          "fontWeight": "bold",
        }
      }
    >
       
      Anglais
    </Text>
  </View>,
  <View>
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 13,
          "lineHeight": 20,
          "marginBottom": 20,
          "marginLeft": 20,
        }
      }
    >
      Changer la langue n'affectera que celle de l'application et non la langue dans laquelle les articles sont affichés, le changement sera à l'échelle du système.
    </Text>
  </View>,
]
`;

exports[`renders correctly 1`] = `
Array [
  <View
    style={
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "backgroundColor": "white",
        "borderColor": "lightgray",
        "borderRadius": 10,
        "borderWidth": 1,
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-around",
        "margin": 10,
        "padding": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 13,
          "fontWeight": "bold",
        }
      }
    >
      Francais
    </Text>
    <RCTSwitch
      accessibilityRole="switch"
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onTintColor="silver"
      style={
        Object {
          "height": 31,
          "width": 51,
        }
      }
      tintColor="silver"
      value={false}
    />
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 13,
          "fontWeight": "bold",
        }
      }
    >
       
      Anglais
    </Text>
  </View>,
  <View>
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 13,
          "lineHeight": 20,
          "marginBottom": 20,
          "marginLeft": 20,
        }
      }
    >
      Changer la langue n'affectera que celle de l'application et non la langue dans laquelle les articles sont affichés, le changement sera à l'échelle du système.
    </Text>
  </View>,
]
`;
